cpp/daal/src/algorithms/svm/svm_train_thunder_impl.i

```diff
@@ -195,28 +162,151 @@ services::Status SVMTrainImpl<thunder, algorithmFPType, cpu>::compute(const Nume
             DAAL_CHECK_STATUS(status, cachePtr->getRowsBlock(wsIndices, nWS, kernelSOARes));
         }
 
-        DAAL_CHECK_STATUS(status, SMOBlockSolver(y, grad, wsIndices, kernelSOARes, nVectors, nWS, cw, eps, tau, buffer.get(), I.get(), alpha,
-                                                 deltaAlpha.get(), diff));
+        DAAL_CHECK_STATUS(status, SMOBlockSolver(y, grad, wsIndices, kernelSOARes, nVectors, nWS, cw, accuracyThreshold, tau, buffer.get(), I.get(),
+                                                 alpha, deltaAlpha.get(), diff));
 
-        DAAL_CHECK_STATUS(status, updateGrad(kernelSOARes, deltaAlpha.get(), gradBuff.get(), grad, nVectors, nWS));
-        if (checkStopCondition(diff, diffPrev, eps, sameLocalDiff) && iter >= nNoChanges) break;
+        DAAL_CHECK_STATUS(status, updateGrad(kernelSOARes, deltaAlpha.get(), gradBuff.get(), grad, nVectors, nTrainVectors, nWS));
+        if (checkStopCondition(diff, diffPrev, accuracyThreshold, sameLocalDiff) && iter >= nNoChanges) break;
         diffPrev = diff;
     }
 
+    if (svmType == SvmType::REGRESSION)
+    {
+        PRAGMA_IVDEP
+        PRAGMA_VECTOR_ALWAYS
+        for (size_t i = 0; i < nVectors; ++i)
+        {
+            alpha[i] = alpha[i] - alpha[i + nVectors];
```

Comment:
Is index check required for SDL?

Reply:
Add check

Reply:
I explicit allocate enough memory myself [here ](https://github.com/oneapi-src/oneDAL/pull/1482/files#diff-5e7f1a748898bf2f0973c524c006103ea510276bd119d90205ac6872c54896a6R92). If we add error, then it would be nice to have a test that could cover this branch of code, but this is impossible. So, I think assertion is quite enough

Reply:
We have the rule for that. Your guess violates rule, right? If it isn't perf-critical we should have an error. Feel free to bring proposal to change the rule at tech.meeting if you have concerns about the rule. You statement about test is unclear: looks like you don't know how to use dependency injection or similar technics to test such a thing.

Reply:
Ok... I made it. My concert - this branch of code will never be reachable. I did it right, the details can be discussed later